include/nbl/video/alloc/SubAllocatedDescriptorSet.h

```diff
@@ -100,39 +141,67 @@ class SubAllocatedDescriptorSet : public core::IReferenceCounted
 
 	// main methods
 
-	//! Warning `outAddresses` needs to be primed with `invalid_value` values, otherwise no allocation happens for elements not equal to `invalid_value`
-	inline void multi_allocate(uint32_t binding, uint32_t count, value_type* outAddresses)
+#ifdef _NBL_DEBUG
+	std::unique_lock<std::recursive_mutex> stAccessVerifyDebugGuard()
 	{
-		#ifdef _NBL_DEBUG
 		std::unique_lock<std::recursive_mutex> tLock(stAccessVerfier,std::try_to_lock_t());
 		assert(tLock.owns_lock());
-		#endif // _NBL_DEBUG
+		return tLock;
+	}
+#else
+	bool stAccessVerifyDebugGuard() { return false; }
+#endif
+
+	//! Warning `outAddresses` needs to be primed with `invalid_value` values, otherwise no allocation happens for elements not equal to `invalid_value`
+	inline void multi_allocate(uint32_t binding, size_type count, value_type* outAddresses)
+	{
+		auto debugGuard = stAccessVerifyDebugGuard();
 
 		auto allocator = getBindingAllocator(binding);
-		for (uint32_t i=0; i<count; i++)
+		for (size_type i=0; i<count; i++)
 		{
 			if (outAddresses[i]!=AddressAllocator::invalid_address)
 				continue;
 
 			outAddresses[i] = allocator->alloc_addr(1,1);
+			// TODO: should also write something to the descriptor set (or probably leave that to the caller?)
 		}
 	}
-	inline void multi_deallocate(uint32_t binding, uint32_t count, const size_type* addr)
+	inline void multi_deallocate(uint32_t binding, size_type count, const size_type* addr)
 	{
-		#ifdef _NBL_DEBUG
-		std::unique_lock<std::recursive_mutex> tLock(stAccessVerfier,std::try_to_lock_t());
-		assert(tLock.owns_lock());
-		#endif // _NBL_DEBUG
+		auto debugGuard = stAccessVerifyDebugGuard();
 
 		auto allocator = getBindingAllocator(binding);
-		for (uint32_t i=0; i<count; i++)
+		for (size_type i=0; i<count; i++)
 		{
 			if (addr[i]==AddressAllocator::invalid_address)
 				continue;
 
 			allocator->free_addr(addr[i],1);
+			// TODO: should also write something to the descriptor sets
```

Comment:
as we discussed we need a new method for the class that handles descriptor set write/copy, to nullify descriptor set tracking

Reply:
basically if you have the `nullDescriptor` device feature enabled, you would indeed write to the DS, else you'd just drop the refcounted references

this needs to be done as a `IGPUDescriptorSet::nullify(const uint32_t binding, std::span<uint32_t> indices)` function + a virtual nullify_impl